cmd/utils.go

```diff
@@ -95,9 +95,55 @@ func splitStr(path, sep string, n int) []string {
 	return splits
 }
 
+func buildS3ConfigFromEnv(alias, urlStr, envConfig string) (*Config, *probe.Error) {
+	// We have a valid alias in env. We populate the
+	// credentials from env.
+	s3Config := new(Config)
+
+	// Fetch keys from the environement, otherwise, get them from the config file
+	keys := splitStr(envConfig, ":", 4)
+	var (
+		accessKey     = keys[0]
+		secretKey     = keys[1]
+		hostURL       = keys[2]
+		signatureType = keys[3]
+	)
+	if !isValidAccessKey(accessKey) || !isValidSecretKey(secretKey) {
+		return nil, probe.NewError(fmt.Errorf("Access/Secret keys %s associated to `%s` "+
+			"are found in your environment but not suitable for use. "+
+			"Falling back to the standard config.", envConfig, alias))
+	}
+	if !isValidHostURL(hostURL) {
+		return nil, errInvalidURL(hostURL)
+	}
+	if signatureType == "" {
+		signatureType = "s3v4"
+	}
+	if signatureType != "" && !isValidAPI(signatureType) {
```

Comment:
signatureType is never empty at this stage, we don't need to do this check `signatureType != ""`